test/distributed/_tensor/test_dtensor.py

```diff
@@ -606,5 +607,33 @@ def test_split_tensor(self) -> None:
                 assert_array_equal(expected_is_tensor_empty, is_tensor_empty)
 
 
+class TestDynamoDTensor(torch._dynamo.test_case.TestCase):
+    def setUp(self):
+        super().setUp()
+        fake_store = FakeStore()
+        dist.init_process_group("fake", store=fake_store, rank=0, world_size=self.world_size)
+
+    @property
+    def device_type(self) -> str:
+        return "cuda" if torch.cuda.is_available() else "cpu"
+
+    @property
+    def world_size(self) -> int:
+        return 2
+
+    def test_dynamo_dtensor(self):
+        mesh = DeviceMesh(self.device_type, torch.arange(self.world_size))
+
+        def fn(x):
+            return x.to_local()
+
+        x = DTensor.from_local(torch.rand(1), mesh, [Replicate()], run_check=False)
```

Comment:
iiuc this test just simulates one rank and i am not sure if DTensor does anything to the data from the torch.rand tensor.  Do we need to at least test some comm operation to make sure dynamo tracing is working?

Reply:
Yep we would need some comm operation to make sure it's getting captured, I'll try to get the `redistribute` call also works with dynamo. What I am not sure yet is whether we expand the `redistribute` call in dynamo or we just make it a `call_method`, and expand that in the aot autograd, probably this would need to be done in aot autograd